DNN module (darknet_io.cpp) compile fails in Visual Studio 12 2013 Win64

System information 
OpenCV => master as at ( https://github.com/opencv/opencv/commit/16821b877cb8bd406b77d822ea35301a492375b2 )
Operating System / Platform => Windows 10 64bit
Compiler => Visual Studio 12 2013 Win64

Detailed description
DNN module (darknet_io.cpp) compile fails in Visual Studio 12 2013 Win64 with the following errors 

17>E:\builds\opencv\source\3.4\modules\dnn\src\darknet\darknet_io.cpp(518): error C2065: 'int32_t' : undeclared identifier
17>E:\builds\opencv\source\3.4\modules\dnn\src\darknet\darknet_io.cpp(518): error C2146: syntax error : missing ';' before identifier 'major_ver'
17>E:\builds\opencv\source\3.4\modules\dnn\src\darknet\darknet_io.cpp(518): error C2065: 'major_ver' : undeclared identifier
17>E:\builds\opencv\source\3.4\modules\dnn\src\darknet\darknet_io.cpp(518): error C2065: 'minor_ver' : undeclared identifier
17>E:\builds\opencv\source\3.4\modules\dnn\src\darknet\darknet_io.cpp(518): error C2065: 'revision' : undeclared identifier
17>E:\builds\opencv\source\3.4\modules\dnn\src\darknet\darknet_io.cpp(519): error C2065: 'major_ver' : undeclared identifier
17>E:\builds\opencv\source\3.4\modules\dnn\src\darknet\darknet_io.cpp(519): error C2065: 'int32_t' : undeclared identifier
17>E:\builds\opencv\source\3.4\modules\dnn\src\darknet\darknet_io.cpp(519): error C2070: 'unknown-type': illegal sizeof operand
17>E:\builds\opencv\source\3.4\modules\dnn\src\darknet\darknet_io.cpp(520): error C2065: 'minor_ver' : undeclared identifier
17>E:\builds\opencv\source\3.4\modules\dnn\src\darknet\darknet_io.cpp(520): error C2065: 'int32_t' : undeclared identifier
17>E:\builds\opencv\source\3.4\modules\dnn\src\darknet\darknet_io.cpp(520): error C2070: 'unknown-type': illegal sizeof operand
17>E:\builds\opencv\source\3.4\modules\dnn\src\darknet\darknet_io.cpp(521): error C2065: 'revision' : undeclared identifier
17>E:\builds\opencv\source\3.4\modules\dnn\src\darknet\darknet_io.cpp(521): error C2065: 'int32_t' : undeclared identifier
17>E:\builds\opencv\source\3.4\modules\dnn\src\darknet\darknet_io.cpp(521): error C2070: 'unknown-type': illegal sizeof operand
17>E:\builds\opencv\source\3.4\modules\dnn\src\darknet\darknet_io.cpp(523): error C2065: 'uint64_t' : undeclared identifier
17>E:\builds\opencv\source\3.4\modules\dnn\src\darknet\darknet_io.cpp(523): error C2146: syntax error : missing ';' before identifier 'seen'
17>E:\builds\opencv\source\3.4\modules\dnn\src\darknet\darknet_io.cpp(523): error C2065: 'seen' : undeclared identifier
17>E:\builds\opencv\source\3.4\modules\dnn\src\darknet\darknet_io.cpp(524): error C2065: 'major_ver' : undeclared identifier
17>E:\builds\opencv\source\3.4\modules\dnn\src\darknet\darknet_io.cpp(524): error C2065: 'minor_ver' : undeclared identifier
17>E:\builds\opencv\source\3.4\modules\dnn\src\darknet\darknet_io.cpp(525): error C2065: 'seen' : undeclared identifier
17>E:\builds\opencv\source\3.4\modules\dnn\src\darknet\darknet_io.cpp(525): error C2065: 'uint64_t' : undeclared identifier
17>E:\builds\opencv\source\3.4\modules\dnn\src\darknet\darknet_io.cpp(525): error C2070: 'unknown-type': illegal sizeof operand
17>E:\builds\opencv\source\3.4\modules\dnn\src\darknet\darknet_io.cpp(528): error C2065: 'int32_t' : undeclared identifier
17>E:\builds\opencv\source\3.4\modules\dnn\src\darknet\darknet_io.cpp(528): error C2146: syntax error : missing ';' before identifier 'iseen'
17>E:\builds\opencv\source\3.4\modules\dnn\src\darknet\darknet_io.cpp(528): error C2065: 'iseen' : undeclared identifier
17>E:\builds\opencv\source\3.4\modules\dnn\src\darknet\darknet_io.cpp(529): error C2065: 'iseen' : undeclared identifier
17>E:\builds\opencv\source\3.4\modules\dnn\src\darknet\darknet_io.cpp(529): error C2065: 'int32_t' : undeclared identifier
17>E:\builds\opencv\source\3.4\modules\dnn\src\darknet\darknet_io.cpp(529): error C2070: 'unknown-type': illegal sizeof operand
17>E:\builds\opencv\source\3.4\modules\dnn\src\darknet\darknet_io.cpp(530): error C2065: 'seen' : undeclared identifier
17>E:\builds\opencv\source\3.4\modules\dnn\src\darknet\darknet_io.cpp(530): error C2065: 'iseen' : undeclared identifier
17>E:\builds\opencv\source\3.4\modules\dnn\src\darknet\darknet_io.cpp(532): error C2065: 'major_ver' : undeclared identifier
17>E:\builds\opencv\source\3.4\modules\dnn\src\darknet\darknet_io.cpp(532): error C2065: 'minor_ver' : undeclared identifier


Steps to reproduce:

1
checkout master 

( as at https://github.com/opencv/opencv/commit/16821b877cb8bd406b77d822ea35301a492375b2 )

2
run cmake as follows:

"C:\Program Files\CMake\bin\cmake.exe" -BE:\builds\opencv\build\ -HE:\builds\opencv\source\3.4\ -G"Visual Studio 12 2013 Win64" -DBUILD_opencv_world=ON -DCUDA_FAST_MATH=ON -DWITH_CUBLAS=ON -DWITH_MKL=ON -DMKL_USE_M
ULTITHREAD=ON -DMKL_WITH_TBB=ON -DWITH_TBB=ON 

3
compile with Visual Studio 12 2013 Win64 (Release, x64) 

cmake log and build log attached for reference

[cmake log.txt](https://github.com/opencv/opencv/files/1606204/cmake.log.txt)
[opencv 3.4 build log.txt](https://github.com/opencv/opencv/files/1606210/opencv.3.4.build.log.txt)

--- a/modules/dnn/src/darknet/darknet_io.cpp
+++ b/modules/dnn/src/darknet/darknet_io.cpp
@@ -66,7 +66,7 @@
 //
 //M*/
 
-#include <opencv2/core.hpp>
+#include "../precomp.hpp"
 
 #include <iostream>
 #include <fstream>

